### PR TITLE
docs: Improve formatting in README

### PR DESCRIPTION
**Pull request by alexandear**
The PR improves formatting in README.md:
- replace outdated `godoc.org` with `pkg.go.dev`;
- set text link for raw URLs;
- replace whitespace with TAB in Go example code;
- add missing `###` for some titles;

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -226,11 +225,11 @@ Alternatively, you can block until the rate limit is reset by using the `context
...
 ```
 
-You can use [go-github-ratelimit](https://github.com/gofri/go-github-ratelimit) to handle
+You can use [gofri/go-github-ratelimit](https://github.com/gofri/go-github-ratelimit) to handle
 secondary rate limit sleep-and-retry for you.
 
-Learn more about GitHub secondary rate limiting at
-https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#about-secondary-rate-limits .
+Learn more about GitHub secondary rate limiting in the
+["About secondary rate limits"](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#about-secondary-rate-limits).
```
Comment (gmlewis) [inline README.md:232]: @Parker77 - what would you recommend for this sentence for proper grammar?
I'm leaning toward `in "About secondary rate limits"` over "at" or "in the".
Thoughts?

File: README.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+[![Go Reference](https://img.shields.io/static/v1?label=godoc&message=reference&color=blue)](https://pkg.go.dev/github.com/google/go-github/v65/github)
 [![Test Status](https://github.com/google/go-github/workflows/tests/badge.svg)](https://github.com/google/go-github/actions?query=workflow%3Atests)
 [![Test Coverage](https://codecov.io/gh/google/go-github/branch/master/graph/badge.svg)](https://codecov.io/gh/google/go-github)
 [![Discuss at go-github@googlegroups.com](https://img.shields.io/badge/discuss-go--github%40googlegroups.com-blue.svg)](https://groups.google.com/group/go-github)
 [![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/796/badge)](https://bestpractices.coreinfrastructure.org/projects/796)
 
 go-github is a Go client library for accessing the [GitHub API v3][].
 
-Currently, **go-github requires Go version 1.13 or greater**.  go-github tracks
+Currently, **go-github requires Go version 1.21 or greater**.  go-github tracks
```
Comment (gmlewis) [inline README.md:12]: So this is interesting because we used to require a minimum of Go 1.13 but at some point we broke that and required a minimum of Go 1.16.

I honestly don't remember now what minimum version of Go is needed but I'm not sure that Go 1.21 is accurate.

Our policy is to officially test the most recent two minor versions of the Go toolchain, so maybe we should just reword it to say that instead of stating an absolute minimum (unless, of course, we can go back and figure out what the absolute minimum needed is).  However, in my many years of maintaining this library, I can't remember once when someone asked to support an older version of the Go toolchain, so maybe this all is moot.

Thoughts?

File: README.md [hunk 3]
```diff
@@ -265,8 +264,8 @@ import "github.com/gregjones/httpcache"
     ).WithAuthToken(os.Getenv("GITHUB_TOKEN"))
 ```
 
-Learn more about GitHub conditional requests at
-https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api?apiVersion=2022-11-28#use-conditional-requests-if-appropriate
+Learn more about GitHub conditional requests in the
+["Use conditional requests if appropriate"](https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api?apiVersion=2022-11-28#use-conditional-requests-if-appropriate).
```
Comment (gmlewis) [inline README.md:268]: Same here... the phrase `in the "Use conditional requests if appropriate"` sounds odd to me.

File: README.md (hunk 1 above)
Comment (Parker77) [reply README.md:232]: I would agree, "in" (with no "the") would be preferable. Same on line 208 above.

File: README.md (hunk 2 above)
Comment (gmlewis) [reply README.md:12]: Maybe "go-github tests against Go version 1.21 and greater" here?

File: README.md [hunk 4]
```diff
@@ -226,11 +225,11 @@ Alternatively, you can block until the rate limit is reset by using the `context
 repos, _, err := client.Repositories.List(context.WithValue(ctx, github.SleepUntilPrimaryRateLimitResetWhenRateLimited, true), "", nil)
 ```
 
-You can use [go-github-ratelimit](https://github.com/gofri/go-github-ratelimit) to handle
+You can use [gofri/go-github-ratelimit](https://github.com/gofri/go-github-ratelimit) to handle
 secondary rate limit sleep-and-retry for you.
 
-Learn more about GitHub secondary rate limiting at
-https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#about-secondary-rate-limits .
+Learn more about GitHub secondary rate limiting in the
```
Comment (gmlewis) [inline README.md:231]:
```suggestion
Learn more about GitHub secondary rate limiting in
```

File: README.md [hunk 5]
```diff
@@ -265,8 +264,8 @@ import "github.com/gregjones/httpcache"
     ).WithAuthToken(os.Getenv("GITHUB_TOKEN"))
 ```
 
-Learn more about GitHub conditional requests at
-https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api?apiVersion=2022-11-28#use-conditional-requests-if-appropriate
+Learn more about GitHub conditional requests in the
```
Comment (gmlewis) [inline README.md:267]:
```suggestion
Learn more about GitHub conditional requests in
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @alexandear !
Just a few nits, please, then we should be ready to merge.

File: README.md (hunk 4 above)
Comment (gmlewis) [reply README.md:231]: WHUPS!!!  It should be Go 1.22 and Go 1.23, so:

Maybe "go-github tests against Go version 1.22 and greater" here?

File: README.md (hunk 2 above)
Comment (alexandear) [reply README.md]: Changed to "go-github tests against Go version 1.22 and greater".

**Review by gmlewis — APPROVED** [review]
Thank you, @alexandear !
LGTM.
Merging.
